actors/common/src/main/java/org/sunbird/learner/util/Util.java

```diff
@@ -1918,6 +1984,124 @@ public static String getCustodianChannel(Map<String, Object> userMap, ActorRef a
     }
     return channel;
   }
+
+  public static void removeEternalFields(
+      String visibilityType, List<String> fieldList, ActorRef actorRef) {
+    Config userProfileConfig = getUserProfileConfig(actorRef);
+    List<String> eternalPublicFields =
+        userProfileConfig.getStringList(JsonKey.ETERNAL_PUBLIC_FIELDS);
+    List<String> eternalPrivateFields =
+        userProfileConfig.getStringList(JsonKey.ETERNAL_PRIVATE_FIELDS);
+    if (JsonKey.PRIVATE.equals(visibilityType)) {
+      for (String field : eternalPrivateFields) {
+        fieldList.remove(field);
+      }
+    } else if (JsonKey.PUBLIC.equals(visibilityType)) {
+      for (String field : eternalPublicFields) {
+        fieldList.remove(field);
+      }
+    }
+  }
+
+  /*
+   * @desc Get user profile configuration from system settings
+   * @param getAllSystemSettings actor reference
+   * @return user profile configuration
+   */
+  public static Config getUserProfileConfig(ActorRef actorRef) {
+    SystemSetting userProfileConfigSetting =
+        getSystemSettingById(JsonKey.USER_PROFILE_CONFIG, actorRef);
+    String userProfileConfigString = userProfileConfigSetting.getValue();
+    Config userProfileConfig = ConfigUtil.parseJSONString(userProfileConfigString);
+    validateUserProfileConfig(userProfileConfig);
+    return userProfileConfig;
+  }
+
+  private static void validateUserProfileConfig(Config userProfileConfig) {
+    if (CollectionUtils.isEmpty(userProfileConfig.getStringList(JsonKey.USER_DATA_FIELDS))) {
```

Comment:
Not all adopters may configure userProfileConfig. Can we have a fallback so existing adopters are not affected.

Reply:
Let us discuss this later today.